src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -621,10 +621,19 @@ def get_user_member_groups(cmd, upn_or_object_id, security_enabled_only=False):
 
 def create_group(cmd, display_name, mail_nickname):
     graph_client = _graph_client_factory(cmd.cli_ctx)
+    matches = list(graph_client.groups.list(filter="displayName eq '{}' and mailNickname eq '{}'".format(
+        display_name, mail_nickname)))
+
+    # workaround to ensure idempotent even AAD graph service doesn't support it
+    if matches:
+        if len(matches) > 1:
+            err = ('There are more than one groups with same display name and mail nick names: "{}". '
```

Comment:
[nit] Grammar is a little off here, I'd have this read: "There is more than one group with the same display name and mail name: {}"